lib/notifications/db/detour.ex

```diff
@@ -33,4 +43,60 @@ defmodule Notifications.Db.Detour do
       :status
     ])
   end
+
+  defmodule Queries do
+    @moduledoc """
+    Defines composable queries for retrieving `Notifications.Db.Detour` info.
+    """
+
+    import Ecto.Query
+
+    @doc """
+    The "base" query that queries `Notifications.Db.Detour`'s without restriction
+    """
+    def base() do
+      from(d in Notifications.Db.Detour, as: :detour_notification, select_merge: d)
+    end
+
+    @doc """
+    Retrieves detour information for notifications from the `Notifications.Db.Detour` table
+    """
+    def get_derived_info(query \\ base()) do
```

Comment:
**Question**: Is this something that could also be used in the `get_detours` endpoint to avoid loading and processing the full JSON blobs from each detour when loading the detour page tables?

Reply:
I think so!

Reply:
I could spend 30m exploring this, do you think it's worth it? I was worried about not having tests for a larger change like that and that does change the footprint, we could do this as a follow-up PR though

Reply:
This was quite interesting to explore, but probably too big for this PR, I'll open it as a follow on.

Reply:
I am very pro follow-up PR for this, and excited that this change will help make some previously-annoying things less annoying!